enclave-manager/web/src/client/enclaveManager/KurtosisClient.ts

```diff
@@ -5,31 +5,51 @@ import {
   DestroyEnclaveArgs,
   EnclaveAPIContainerInfo,
   EnclaveInfo,
-  EnclaveMode,
+  EnclaveMode
 } from "enclave-manager-sdk/build/engine_service_pb";
 import { KurtosisEnclaveManagerServer } from "enclave-manager-sdk/build/kurtosis_enclave_manager_api_connect";
 import {
   GetListFilesArtifactNamesAndUuidsRequest,
   GetServicesRequest,
   GetStarlarkRunRequest,
-  RunStarlarkPackageRequest,
+  RunStarlarkPackageRequest
 } from "enclave-manager-sdk/build/kurtosis_enclave_manager_api_pb";
 import { assertDefined, asyncResult } from "../../utils";
 import { RemoveFunctions } from "../../utils/types";
 
 export abstract class KurtosisClient {
   protected readonly client: PromiseClient<typeof KurtosisEnclaveManagerServer>;
-  protected readonly browserBasePathUrl: string;
 
-  constructor(client: PromiseClient<typeof KurtosisEnclaveManagerServer>, browserBasePathUrl: string) {
+  /* Full URL of the browser containing the EM UI covering two use cases:
+   * In local-mode this is: http://localhost:9711, http://localhost:3000 (with `yarn start` / dev mode)
+   * In authenticated mode this is: https://cloud.kurtosis.com/enclave-manager (this data/url is provided as a search param when the code loads)
+   *
+   * This URL is primarily used to generate links to the EM UI (where the hostname is included).
+   * */
+  protected readonly parentUrl?: URL;
+
+  /* Full URL of the EM UI, covering two use cases:
+   * In local-mode this is the same as the `parentUrl`
+   * In authenticated mode : https://cloud.kurtosis.com/enclave-manager/gateway/ips/1-2-3-4/ports/1234/?searchparams... (this data/url is provided as a search param when the code loads)
+   *
+   * This URL is primarily used to set the react router basename so that the router is able to ignore any leading subdirectories.
+   * */
+  protected readonly childUrl?: URL;
```

Comment:
nbd, but knowing `childUrl` and `parentUrl` is not the responsibility of the KurtosisClient (which should only be concerned with talking to an enclave manager backend).

It would be more appropriate to just store these values in state inside the KurtosisClientContextProvider, and have a way of accessing them from there.